### PR TITLE
Update trustping.md

### LINKED ISSUE #271: Trust Ping protocol needs a `from` attribute
In order to respond to a trust ping the receiver needs to know where to send the response. Is there a different way to determine the sender that I'm not aware of?

### PR DESCRIPTION
**Pull request by brianorwhatever**
closes #271

### CODE REVIEW

**Comment by TelegramSam** [comment]
I vote that `from` is required for this protocol, even if no response is required. The point of the ping is to verify a trusted relationship. If no `from`, no relationship.

**Comment by TelegramSam** [comment]
> How is identifier authentication being done now? Couldn't we rely on that methodology to get it instead?

You mean authenticated encryption? That is the method I'm expecting we rely on.

**Comment by brianorwhatever** [comment]
So I think the issue here is that the "Trust Ping" protocol becomes just the "Ping" protocol when completed using anonymous encryption. I don't believe there is a way to authenticate in this case and all we will be able to rely on is a self declared `from` value

**Comment by swcurran** [comment]
If you just send out a trust ping to a previously unknown contact using anoncrypt, yes.  But you can use an existing connection and (I think, if I've been following the conversation), the message would be delivered using authcrypt.

I don't see a need to change the name.  The information about using the DIDComm 2 with and without a connection should be general information and need not be tied to each protocol.

**Comment by TelegramSam** [comment]
I've proposed in #274 that we make from required. That clears up several things, including this confusion about a trustpring without a from. Thoughts?

File: docs/spec-files/trustping.md
Comment (awoie) [inline docs/spec-files/trustping.md:59]: @brianorwhatever I guess we can remove the `from` after we merged #274

**Review by awoie — CHANGES_REQUESTED** [review]
@brianorwhatever I guess we can remove the from after we merged #274

File: docs/spec-files/trustping.md
Comment (awoie) [inline docs/spec-files/trustping.md:59]:
```suggestion
```

**Review by awoie — CHANGES_REQUESTED** [review]
lgtm after removing the `from` language, but merge should wait until #274 got merged

File: docs/spec-files/trustping.md
Comment (brianorwhatever) [reply docs/spec-files/trustping.md]: thanks @awoie

**Comment by kdenhartog** [comment]
> > 
> 
> You mean authenticated encryption? That is the method I'm expecting we rely on.

Yup, I've started to rephrase my language to prevent conflation between what we mean when we say "authenticated encryption"/Authcrypt and what's more commonly used in symmetrical encryption where they're describing a specific security property that's slightly different than what we mean.

In my mind identifier authentication utilizes traditional "authenticated encryption" (they normally just mean the encryption is integrity protected and tamper resistant by people without the secret key) and that we're binding the integrity to a specific identifier (e.g. the did) that can be inferred. This may not be exactly correct terminology, but just looking to stray away from conflating language when describing different security properties. Let's open a separate issue to discuss this further if you want to just so that we don't slow down this PR from being merged.

**Pull request merged**
